cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -601,7 +601,9 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
                                 clusterCaConfig == null || clusterCaConfig.isGenerateCertificateAuthority(),
                                 clusterCaConfig != null ? clusterCaConfig.getCertificateExpirationPolicy() : null);
                         clusterCa.createRenewOrReplace(
-                                reconciliation.namespace(), reconciliation.name(), caLabels.toMap(),
+                                reconciliation.namespace(), reconciliation.name(),
+                                AbstractModel.mergeLabelsOrAnnotations(caLabels.toMap()
+                                ,kafkaAssembly.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata().getLabels()),
```

Comment:
The way this is implemented, the metadata from the template will be applied to both secrets:
* The one with the public key
* The one with the private key

Is this really what you want? If you use this to trigger copying of the secret, I assume it will copy both of them and as a result it will basically leak the private key to the other namespaces where it can be misused. 

So that does not seem to be desired to me. I think it would be better to just pass the value here as another parameter and use it only for the secret with the public key (i.e. call `mergeLabelsOrAnnotations` only inside the `createRenewOrReplace` for the public key).

In case I'm wrong and this is desired, I think you should name the field in the `api` class just `clusterCa` and not `
clusterCaCert`. But I think it is preferred to have the labels and annotations only for the public key secret.